addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScanner.java

```diff
@@ -315,28 +365,100 @@ private ApplicationMatch addIfDomMatches(
         return appMatch;
     }
 
-    private ApplicationMatch addIfMatches(
+    private static ApplicationMatch addIfMatches(
             ApplicationMatch appMatch,
             Application currentApp,
             AppPattern appPattern,
             String content) {
-        List<String> results = appPattern.findInString(content);
-        if (results != null) {
+        Result result = appPattern.findInString(content);
+        if (!result.getVersions().isEmpty() || !result.getEvidence().isEmpty()) {
             appMatch = getAppMatch(appMatch, currentApp);
             // TODO may need to account for the wappalyzer spec in dealing with version info:
             // https://www.wappalyzer.com/docs/specification
-            results.forEach(appMatch::addVersion);
+            appMatch.addEvidence(result.getEvidence());
+            result.getVersions().forEach(appMatch::addVersion);
```

Comment:
Be good to also have tests for these changes.

Reply:
What kind of thing are you expecting here?

Reply:
I've separated the tests a bit and updated the assertions to ensure there's always evidence when expected. Versions were already being checked in applicable cases.

Reply:
The `PassiveScannerAlertsUnitTest` is not really needed, if that's to group them add them as an inner class.

Reply:
Moved as `@Nested`